src/main/java/org/candlepin/subscriptions/tally/TallyTaskFactory.java

```diff
@@ -66,7 +67,14 @@ public TallyTaskFactory(Validator validator, TallySnapshotController snapshotCon
   @Override
   public Task build(TaskDescriptor taskDescriptor) {
     if (taskDescriptor.getTaskType() == TaskType.UPDATE_SNAPSHOTS) {
-      return new UpdateAccountSnapshotsTask(snapshotController, taskDescriptor.getArg("accounts"));
+      if (taskDescriptor.hasArg("orgs")) {
+        log.debug("Task created for processing orgs");
+        return new UpdateOrgSnapshotsTask(snapshotController, taskDescriptor.getArg("orgs"));
+      } else {
+        log.debug("Task created for processing accounts");
+        return new UpdateAccountSnapshotsTask(
+            snapshotController, taskDescriptor.getArg("accounts"));
```

Comment:
I'd rather throw an Exception here with the taskDescriptor as part of the message, since we won't be producing task messages with accounts anymore after this (see later comment in CaptureSnapshotsTaskManager.java). And then we should remove UpdateAccountSnapshotsTask altogether).

Reply:
Done